test/app/core/managers/audio_record_services_test.dart

```diff
@@ -0,0 +1,313 @@
+        when(() => audioSyncManager.syncAudio()).thenAnswer((_) async => true);
+
+        // act
+        await recordServices.stop();
+
+        // assert
+        verify(() => soundRecorder.isStopped).called(1);
+        verify(() => soundRecorder.stopRecorder()).called(1);
+      },
+    );
+
+    test(
+      'stop() do not crash if soundRecorder crash',
+      () async {
+        // arrange
+        when(() => soundRecorder.isStopped).thenReturn(false);
+        when(() => soundRecorder.stopRecorder()).thenThrow(Exception());
+
+        // act / assert
+        await expectLater(
+          recordServices.stop(),
+          completion(isNull),
+        );
+      },
+    );
+
+    test(
+      'onProgress shows the record progress',
+      () async {
+        // arrange
+        final mockPermissionHandlerPlatform =
+            PermissionHandlerPlatform.instance;
+        debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+
+        when((() => mockPermissionHandlerPlatform
+                .checkPermissionStatus(Permission.microphone)))
+            .thenAnswer((_) async => PermissionStatus.granted);
+
+        when(() => soundRecorder.isStopped).thenReturn(true);
+        when(() => soundRecorder.openRecorder()).thenAnswer((_) async => null);
+        when(() => soundRecorder.setSubscriptionDuration(any()))
+            .thenAnswer((_) async => null);
+        when(() => soundRecorder.startRecorder(
+              codec: any(named: 'codec'),
+              toFile: any(named: 'toFile'),
+              bitRate: any(named: 'bitRate'),
+              sampleRate: any(named: 'sampleRate'),
+            )).thenAnswer((_) async => null);
+
+        when(() => audioSyncManager.audioFile(
+              session: any(named: 'session'),
+              sequence: any(named: 'sequence'),
+            )).thenAnswer((_) async => '/tmp/file_name.txt');
+
+        when(() => soundRecorder.onProgress).thenAnswer(
+          (_) => Stream.fromIterable(
+            [RecordingDisposition(Duration(minutes: 1), 50.0)],
+          ),
+        );
+
+        // act
+        await recordServices.start();
+        final result = await recordServices.onProgress.first;
+
+        // assert
+        expect(result, isA<AudioActivity>());
+        expect(result.decibels, 50.0);
+        expect(result.time, '01:00:00');
+      },
+    );
+
+    test('rotate() trigger action to rotate the audio resource', () async {
+      // arrange
+      final mockPermissionHandlerPlatform = PermissionHandlerPlatform.instance;
+      debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+
+      when((() => mockPermissionHandlerPlatform
+              .checkPermissionStatus(Permission.microphone)))
+          .thenAnswer((_) async => PermissionStatus.granted);
+
+      when(() => soundRecorder.isStopped).thenReturn(true);
+      when(() => soundRecorder.openRecorder()).thenAnswer((_) async => null);
```

Comment:
O retorno do método `flutterSoundPlayer.openPlayer()` é `Future<void>`, nunca retornará nulo, acha que faria sentido o mock ser `void` também?
```suggestion
      when(() => soundRecorder.openRecorder()).thenAnswer((_) async {});
```